modules/rostests/kmtests/ntos_ex/ExResource.c

```diff
@@ -193,21 +196,42 @@ TestResourceUndocumentedShortcuts(
     /* ExEnterCriticalRegionAndAcquireResourceShared, ExEnterCriticalRegionAndAcquireSharedWaitForExclusive */
     Count = 0;
     Ret = pExEnterCriticalRegionAndAcquireResourceShared(Res); ++Count;
-    ok_eq_pointer(Ret, KeGetCurrentThread()->Win32Thread);
+    if (GetNTVersion() <= _WIN32_WINNT_WS03)
+    {
+        ok_eq_pointer(Ret, KeGetCurrentThread()->Win32Thread);
+    }
+    else
+    {
+        ok_eq_pointer(Ret, NULL);
+    }
```

Comment:
Can be simplified to:
```suggestion
    ok_eq_pointer(Ret, GetNTVersion() <= _WIN32_WINNT_WS03 ? KeGetCurrentThread()->Win32Thread : NULL);
```
(and in some other places)